lib/sle2docker/native_image.rb

```diff
@@ -34,35 +34,36 @@ def activate
     private
 
     def compute_metadata_file
+      match = parse_image_filename(@image_name)
+      @metadata = parse_metadata_file("#{match['metadata_file']}.metadata")
+
+      @repository   = @metadata['image']['name']
+      @tag          = @metadata['image']['tags'][0]
+      @build        = match['build']
+      @image_id     = "#{@repository}:#{@tag}"
+      @options['tag_with_build'] && @image_id << '-' + @build
+    end
+
+    def parse_image_filename(file)
       # example of image name and metadata file:
       # kiwi >= 8.30
       #      sles12sp3-container.x86_64-2.0.1-Build2.3.docker (image basename)
       #      sles12sp3-container.x86_64-2.0.1.metadata
       regexp = /(?<metadata_file>.*\d+\.\d+\.\d+)
         (-Build(?<build>\d+\.\d+)\.docker)?/x
-      match = regexp.match(@image_name)
+      match = regexp.match(file)
       match.nil? &&
         raise(DockerTagError,
-              "Docker image #{@image_name} not found. "\
+              "Docker image #{file} not found. "\
               'Run sle2docker list to check which docker images are available.')
-      @metadata = parse_metadata_file("#{match['metadata_file']}.metadata")
-
-      validate_metadata(@metadata)
-
-      @repository   = @metadata['image']['name']
-      @tag          = @metadata['image']['tags'][0]
-      @build        = match['build']
-      @image_id     = "#{@repository}:#{@tag}"
+      match
     end
 
-    def parse_metadata_file(metadata)
+    def parse_metadata_file(metadata_file)
       file = File.read(
-        File.join(NativeImage::IMAGES_DIR, metadata)
+        File.join(NativeImage::IMAGES_DIR, metadata_file)
       )
-      JSON.parse(file)
-    end
-
-    def validate_metadata(metadata)
+      metadata = JSON.parse(file)
       metadata['image']['tags'][0].to_s.empty? &&
         raise(DockerTagError,
```

Comment:
An exception is raised if not tag, also, 'latest' is always one of the tags present, see line 29. Also this tag should no be a guess in any case, as it is used to validate if the current image is activated or not.